CMakeLists.txt

```diff
@@ -1,16 +1,22 @@
-cmake_minimum_required(VERSION 2.8)
-project(azmq)
+cmake_minimum_required(VERSION 3.14)
+project(azmq VERSION 1.0.3 LANGUAGES CXX)
 
-include(AzmqCPack.cmake)
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 
-set(CMAKE_CXX_STANDARD 11)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_EXTENSIONS OFF)
-set(CMAKE_XCODE_ATTRIBUTE_CLANG_CXX_LANGUAGE_STANDARD "c++11")
+include(GNUInstallDirs)
+include(AzmqCPack)
 
-if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
-endif()
+find_package(Boost 1.48 COMPONENTS system date_time thread chrono random REQUIRED)
+find_package(ZeroMQ 4.0 REQUIRED)
```

Comment:
In cppzmq, they modified their CMakeLists.txt to not call find_package if the target already exists.  This approach better works if the master project is using CMake's FetchContent* calls.  I really like all the changes in this PR and was wondering if something like this could get added?

i.e. 
```
if (NOT TARGET libzmq AND NOT TARGET libzmq-static)
  find_package(ZeroMQ QUIET)
  ...
endif()
```